Supply the version of LLVM we are intending to use via QDOC_USER_CLANG_VERSION,
a made-up CMake varible derived from LLVM_VERSION in the port Makefile, instead
of iterating through potential candidates. This avoids mismatches during the
configure stage when a user has a version of LLVM installed greater than what is
set in DEFAULT_VERSIONS.

PR: 287430

--- .cmake.conf.orig	2025-05-29 00:19:42 UTC
+++ .cmake.conf
@@ -3,7 +3,7 @@ set(QDOC_SUPPORTED_CLANG_VERSIONS
 set(QT_EXTRA_INTERNAL_TARGET_DEFINES "QT_NO_AS_CONST=1")
 set(QDOC_MINIMUM_CLANG_VERSION "17")
 set(QDOC_SUPPORTED_CLANG_VERSIONS
-    "20.1" "19.1" "18.1" "17.0"
+    ${QDOC_USER_CLANG_VERSION}
 )
 list(APPEND QT_EXTRA_INTERNAL_TARGET_DEFINES "QT_NO_FOREACH=1")
 list(APPEND QT_EXTRA_INTERNAL_TARGET_DEFINES "QT_NO_CONTEXTLESS_CONNECT=1")
